flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/sql/JoinITCase.scala

```diff
@@ -83,6 +83,27 @@ class JoinITCase(
     val env = ExecutionEnvironment.getExecutionEnvironment
     val tEnv = TableEnvironment.getTableEnvironment(env, config)
 
+    val sqlQuery = "SELECT c, g FROM Table3, Table5 WHERE b = e AND a < 6"
```

Comment:
I think this test covers the same case as the one above (`testJoinWithFilter()`).